examples_tests/10.AllocatorTest/main.cpp

```diff
@@ -0,0 +1,478 @@
+
+	inline uint32_t getRandomNumber(uint32_t rangeBegin, uint32_t rangeEnd)   
+	{
+		std::uniform_int_distribution<uint32_t> dist(rangeBegin, rangeEnd);
+		return dist(mt);
+	}
+
+private:
+	std::mt19937 mt;
+	const std::uniform_int_distribution<uint32_t> allocsPerFrameRange = std::uniform_int_distribution<uint32_t>(minAllocs, maxAllocs);
+	const std::uniform_int_distribution<uint32_t> sizePerFrameRange = std::uniform_int_distribution<uint32_t>(1u, maxSizeForSquaring);
+	const std::uniform_int_distribution<uint32_t> maxAlignmentExpPerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxAlignmentExp);
+
+	const std::uniform_int_distribution<uint32_t> buffSzRange = std::uniform_int_distribution<uint32_t>(minVirtualMemoryBufferSize, maxVirtualMemoryBufferSize);
+	const std::uniform_int_distribution<uint32_t> offsetPerFrameRange = std::uniform_int_distribution<uint32_t>(0u, maxOffset);
+	const std::uniform_int_distribution<uint32_t> alignOffsetPerFrameRange = std::uniform_int_distribution<uint32_t>(0u, maxAlignOffset);
+	const std::uniform_int_distribution<uint32_t> blockSizePerFrameRange = std::uniform_int_distribution<uint32_t>(1, maxBlockSize);
+};
+
+template<typename AlctrType>
+class AllocatorHandler
+{
+	using Traits = core::address_allocator_traits<AlctrType>;
+public:
+	
+	AllocatorHandler(AllocationCreationParameters& allocationCreationParameters)
+		: creationParameters(allocationCreationParameters) 
+	{
+	}
+	
+	void executeAllocatorTest()
+	{
+		for (size_t i = 0; i < creationParameters.multiAllocCnt; ++i)
+			executeForFrame();
+	}
+
+private:
+	struct AllocationData
+	{
+		uint32_t outAddr = AlctrType::invalid_address;
+		uint32_t size = 0u;
+		uint32_t align = 0u;
+	};
+
+	struct RandParams
+	{
+		uint32_t maxAlign;
+		uint32_t addressSpaceSize;
+		uint32_t alignOffset;
+		uint32_t offset;
+		uint32_t blockSz;
+	};
+
+private:
+	void executeForFrame()
+	{
+		RandParams randAllocParams = getRandParams();
+		void* reservedSpace = nullptr;
+		AlctrType alctr;
+
+		if constexpr (std::is_same<AlctrType, core::LinearAddressAllocator<uint32_t>>::value)
+		{
+			alctr = AlctrType(nullptr, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize);
+		}
+		else
+		{
+			const auto reservedSize = AlctrType::reserved_size(randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+			reservedSpace = _IRR_ALIGNED_MALLOC(reservedSize, _IRR_SIMD_ALIGNMENT);
+			alctr = AlctrType(reservedSpace, randAllocParams.offset, randAllocParams.alignOffset, randAllocParams.maxAlign, randAllocParams.addressSpaceSize, randAllocParams.blockSz);
+		}
+
+		// randomly decide how many `multi_allocs` to do
+		const uint32_t multiAllocCnt = rng.getRandomNumber(1u, 5u); //TODO: will change it later
+		for (uint32_t i = 0u; i < multiAllocCnt; i++)
+		{
+			outAddresses.clear();
+			sizes.clear();
+			alignments.clear();
```

Comment:
this is wrong, it means you deallocate all allocations from each multi_alloc (which means you'll never multi_alloc on a fragmented allocator)

Reply:
my bad, it got this confused with `results`